### DISCUSSION TITLE
Powershell Delete Reservation

### DISCUSSION THREAD

**Discussion by jkieffer0616** [Azure/ipam#93] [Q&A] [answered]
Can anyone help me with an example of deleting a reservation using powershell?  

I have been trying different variations of the following but have had no success.

$spaceName="SpaceName"
$blockName="BlockName"
$apiURL="https://IPAM-APP-SERVICE.azurewebsites.net/api/spaces/$spaceName/blocks/$blockName/reservations"
$headers = @{Authorization = "Bearer $Token"} 
$contentType = "application/json"
$methodType="Delete"

$reservationID="GQdjUst2gd5tkQBg5ve2ka"
$JSONBody=$reservationID | ConvertTo-Json

$response=Invoke-RestMethod -Method $methodType -Body $JSONBody -uri $apiURL -Headers $headers -ContentType "$contentType"

This returns the error:  value is not a valid list","type":"type_error.list

Any help is appreciated.  Thanks!

**Comment by DCMattyG** [comment, marked as answer]
Hi @jkieffer0616, if you take a look at the Swagger docs, you'll notice that the body needs to be an *ARRAY* of Reservation ID's. This is true even for a single Reservation you'd like to delete.

![image](https://user-images.githubusercontent.com/12955732/226985569-ba831520-f56d-417c-b8e5-492b70ac4e07.png)

That said, you can use this format for the body:

```
$reservationID = @("GQdjUst2gd5tkQBg5ve2ka")
$JSONBody = $reservationID | ConvertTo-Json -AsArray
```

Note that you'll want to use the **-AsArray** switch because when you have a single item in an Array, PowerShell will present it as a string instead of an Array (which is not what we want).

P.S. I'm sure you've seen it already, but we have our documentation with example API calls via PowerShell [here](https://azure.github.io/ipam/#/api/README?id=sample-api-calls)

Hope that helps, if not just let me know and I'm happy to work with you until everything is working just as you'd expect 😃

**Reply by jkieffer0616** [reply]
Thank you so much for the quick help!  It was the missing -AsArray switch that was causing my issues.

**Reply by DCMattyG** [reply]
My absolute pleasure. Please let us know if you need anything else!
